### PR TITLE
Updates related to upf-adapter

### PR DESCRIPTION
**Pull request by gab-arrobo**
This PR includes only changes related to the upf-adapter:

- Remove Dockerfile
- Bump up GO to 1.21
- Add action for building upfadapter

### CODE REVIEW

**Comment by gab-arrobo** [comment]
ok to test

File: upfadapter/Dockerfile [hunk 1]
```diff
@@ -3,26 +3,25 @@
...
-FROM alpine
+# Install debug tools ~ 50MB (if DEBUG_TOOLS is set to true)
+RUN if [ "$DEBUG_TOOLS" = "true" ]; then \
+        apk update && apk add -U vim strace net-tools curl netcat-openbsd bind-tools bash; \
+        fi
 
 WORKDIR /bin
 
-COPY --from=builder /app/upf-adapter /bin/
+COPY --from=builder /app/upf-adapter .
```
Comment (thakurajayL) [inline upfadapter/Dockerfile:25]: Did you try to run to see if upfadapter comes up in AIAB ? enable upf adapter in the override file.

Comment (gab-arrobo) [reply upfadapter/Dockerfile:25]: No, I have not tried. Will do it later today and get back to you.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -24,6 +24,20 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        working-directory: ./upfadapter
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build
+        working-directory: ./upfadapter
```
Comment (gab-arrobo) [inline .github/workflows/master.yml:38]:
```suggestion
    defaults:
      run:
        working-directory: ./upfadapter
    steps:
      - uses: actions/checkout@v4

      - uses: actions/setup-go@v5
        with:
          go-version-file: 'go.mod'

      - name: Build
```

**Review by gab-arrobo — COMMENTED** [review]
An improvement

**Comment by gab-arrobo** [comment]
@thakurajayL, now we are ready to merge this PR (the Jenkins checks were removed)

File: .github/dependabot.yml [hunk 3]
```diff
@@ -19,3 +19,8 @@ updates:
     directory: "/"
     schedule:
       interval: "daily"
+
+  - package-ecosystem: "gomod"
+    directory: "/upfadapter/"
+    schedule:
+      interval: "daily"
```
Comment (thakurajayL) [inline .github/dependabot.yml:26]: You wanted it weekly ?

Comment (gab-arrobo) [reply .github/dependabot.yml]: Updated it. Thanks!

**Review by thakurajayL — APPROVED** [review]

**Pull request merged**